src/D2L.Bmx/ConsolePrompter.cs

```diff
@@ -32,8 +46,27 @@ internal static class ConsolePrompter {
 		return user;
 	}
 
-	public static int? PromptDefaultDuration() {
-		Console.Write( "Default duration of session in minutes (optional, default: 60): " );
+	string IConsolePrompter.PromptPassword() {
+		Console.Error.Write( "Okta Password: " );
+		string password = "";
+
+		// TODO: remove nomask
+		while( true ) {
+			var input = Console.ReadKey( intercept: true );
+			if( input.Key == ConsoleKey.Enter ) {
+				Console.Error.Write( '\n' );
+				break;
+			} else if( input.Key == ConsoleKey.Backspace && password.Length > 0 ) {
```

Comment:
IDE isn't a fan of this if else chain here, nor the one in ``PromptMfa``.

It does feel like a clearer read to me if it is separate, or alternatively like:
https://github.com/Brightspace/bmx/blob/46f5346c6b1be35e158d8366bedaf96b43980c07/Bmx.CommandLine/CommandLine.cs#L102-L112

WDYT?

Reply:
What's the IDE's suggestion here?
I pretty often their suggestions on if else vs expression for example, because I often find them just bad...

Reply:
```csharp
if( input.Key == ConsoleKey.Enter ) {
	Console.Error.Write( '\n' );
	break;
}

if( input.Key == ConsoleKey.Backspace && password.Length > 0 ) {
	password = password.Remove( password.Length - 1, 1 );
} else if( !char.IsControl( input.KeyChar ) ) {
	password += input.KeyChar;
}
```

I do like it in the context of loop like this to have exit conditions separate from the rest. 

> I'm a little reluctant to do more cursor manipulation here...

Just the do while loop over the input to be clear (move the exit condition out of the loop body)

Reply:
oh I see. I do like early exit.
there's quite a few places in bmx that can be updated for early exit.
I'll update these later.

I have another change lined up to change the whole password input thing. constructing a new string on every key stroke is bad form in itself...